base/asyncmap.jl

```diff
@@ -3,161 +3,336 @@
 using Base.Iterators.Enumerate
 
 """
-    AsyncCollector(f, results, c...; ntasks=0) -> iterator
-
-Apply `f` to each element of `c` using at most `ntasks` asynchronous
-tasks.
-If `ntasks` is unspecified, uses `max(100, nworkers())` tasks.
-For multiple collection arguments, apply `f` elementwise.
-Output is collected into `results`.
+    asyncmap(f, c...; ntasks=0, batch_size=nothing)
+
+Uses multiple concurrent tasks to map `f` over a collection (or multiple
+equal length collections). For multiple collection arguments, `f` is
+applied elementwise.
+
+`ntasks` specifies the number of tasks to run concurrently.
+Depending on the length of the collections, if `ntasks` is unspecified,
+upto a 100 tasks will be used for concurrent mapping.
+
+`ntasks` can also be specified as a zero-arg function. In this case, the
+number of tasks to run in parallel is checked before processing every element and a new
+task started if the value of `ntasks_func()` is less than the current number
+of tasks.
+
+If `batch_size` is specified, the collection is processed in batch mode. `f` should
+then be a function that should accept a `Vector` of argument tuples and should
+return a vector of results. The input vector will have a length upto `batch_size`
+
+The following examples return the `object_id` of the tasks in which the mapping
+function is executed for different invocations of `ntasks` and `batch_size`.
+
+With ntasks undefined, each element is processed in a different task.
+```
```

Comment:
I considered that. Since the examples are printing out `object_id`s, wouldn't the doctest fail?